clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
@hachikuji I appreciate your taking time to discuss these in detail. Here is my thought:

1. I am OK with the solution you have proposed. My understanding is that the performance regression exists but is negligible. I am resisting your attempts to address these shortcomings because performance improvement with the fix is even smaller than the performance hits. I think my logic is consistent here, right? Since we both agree that the performance hit is not an issue, why do we both to add new methods to address any problem?

I sincerely believe that the performance hit with your approach is negligible. Can we just use the existing patch without adding that new method? But if you insistent, I will just add it.

2. I haven't tried that because I thought we have reached agreement that the current approach can work. As I said I just want to fix this problem and wouldn't any negligible minor performance hit. Thus I prefer to add that new method if you want me to do it, instead of re-writing a new approach.